### ISSUE TITLE
About the result of the training

### ISSUE THREAD

**Issue by ZhengDonglei** [Beniko95J/MLF-VO#6]
Dear officer:
    It seems that the final train result can t reach the result you mentioned in the paper and also the master branch .I train the model in the same manner using NVIDIA RTX 3090 and other option are the same. My question is can you provide more detail about how to training?

**Comment by Beniko95J** [comment]
Hi, thank you for interests.

The settings in the `dev` branch should be right and I think the only thing needed to change is the selected pose network.
May I ask that have you checked the curves of `test/t_rel` and `test/r_rel` in tensorboard logs? The best model may not be the last one, and we choose the best model from the last 5 epochs according to the curves of `test/t_rel` and `test/r_rel` in our experiments.

**Comment by SBaokun** [comment]
![image](https://github.com/Beniko95J/MLF-VO/assets/52092469/66a5780d-b2b2-47ae-af31-633b2a5c3b1e)
![image](https://github.com/Beniko95J/MLF-VO/assets/52092469/28e75e03-acaf-4983-8c19-54e8f6f7b3df)
thank a lot for your work! i train the model using 4090,but i find the final train result can t reach the result in your paper.especially, ATE in suquence 9.Can you give me some suggestion?thanks!

**Comment by Beniko95J** [comment]
It seems that the training diverges during the last epochs. The performance on Seq.09 is getting worse than the epochs before 35.You can try to decrease the learning rate again in the last 10 epochs (I remember that the default setting is 1e-4 for epoch 1~20 and 5e-5 for epoch 21~40) and search the best model in a larger range. Since the ATE largely depends on the rotation estimation, you can seach the best model based on the number of `average rotational error`. However, the ATE performance may still be very sensittive, which is similarily reported in [here](https://github.com/JiawangBian/SC-SfMLearner-Release/issues/58) and [here](https://github.com/JiawangBian/SC-SfMLearner-Release/issues/73).

**Comment by SBaokun** [comment]
Thank you for the quick reply. I have adjusted lr and results have improved.

**Comment by SBaokun** [comment]
> It seems that the training diverges during the last epochs. The performance on Seq.09 is getting worse than the epochs before 35.You can try to decrease the learning rate again in the last 10 epochs (I remember that the default setting is 1e-4 for epoch 1--20 and 5e-5 for epoch 21--40) and search the best model in a larger range. Since the ATE largely depends on the rotation estimation, you can seach the best model based on the number of `average rotational error`. However, the ATE performance may still be very sensittive, which is similarily reported in [here](https://github.com/JiawangBian/SC-SfMLearner-Release/issues/58) and [here](https://github.com/JiawangBian/SC-SfMLearner-Release/issues/73).

Thank you for the quick reply. I have adjusted lr and results have improved.

**Comment by SBaokun** [comment]
I am sorry to interrupt again. 
When I train the multi_layer model, the eval results are the same as when  with the master branch(test use run_odometry.py). But when I train the middle  model , the two results are very different. Can you offer some advice?
another question is networks/sparate_model == model/multi_layer_fusion.py? 

Visualisation of the training process
![image](https://github.com/Beniko95J/MLF-VO/assets/52092469/f044b37d-92be-4e41-806a-2ea0f74422d3)

result use run_odometry.py in master branch
![image](https://github.com/Beniko95J/MLF-VO/assets/52092469/31e6daac-4a1b-4c05-bdc9-74f21660c1e0)
